WordPress/Classes/Services/ReaderTopicService.m

```diff
@@ -362,6 +366,42 @@ - (ReaderTopic *)siteTopicForPost:(ReaderPost *)post
     return topic;
 }
 
+- (void)siteTopicForSiteWithID:(NSNumber *)siteID
+                       success:(void (^)(NSManagedObjectID *objectID, BOOL isFollowing))success
+                       failure:(void (^)(NSError *error))failure
+{
+    ReaderTopicServiceRemote *remoteService = [[ReaderTopicServiceRemote alloc] initWithApi:[self apiForRequest]];
+    [remoteService fetchSiteInfoForSiteWithID:siteID success:^(RemoteReaderSiteInfo *siteInfo) {
+        if (!success) {
+            return;
+        }
+        NSEntityDescription *entity = [NSEntityDescription entityForName:NSStringFromClass([ReaderTopic class])
+                                                  inManagedObjectContext:self.managedObjectContext];
+        ReaderTopic *topic = [[ReaderTopic alloc] initWithEntity:entity
+                     insertIntoManagedObjectContext:self.managedObjectContext];
+
+        topic.topicID = siteInfo.siteID;
+        topic.type = ReaderTopicTypeSite;
+        topic.title = siteInfo.siteName;
+        topic.topicDescription = siteInfo.siteDescription;
+        topic.path = [NSString stringWithFormat:@"%@sites/%@/posts/", WordPressRestApiEndpointURL, siteInfo.siteID];
+
+        NSError *error;
+        [self.managedObjectContext obtainPermanentIDsForObjects:@[topic] error:&error];
+        if (error) {
+            DDLogError(@"%@ error obtaining permanent ID for topic for site with ID %@: %@", NSStringFromSelector(_cmd), siteID, error);
+        }
+        success(topic.objectID, siteInfo.isFollowing);
```

Comment:
Probably lame question: the **save** call was expected to be executed by the caller right?